hrc_speech_prediction/models.py

```diff
@@ -90,12 +150,183 @@ def fit(self, X_context, X_speech, labels):
         super(PragmaticModel, self).fit(X_context, X_speech, labels)
         self._p = X_context.sum(axis=0) + self.alpha
         self._p /= self._p.sum()
-        probas = super(PragmaticModel, self)._predict_proba(X_context, X_speech)
+        probas = super(PragmaticModel, self)._predict_proba(
+            X_context, X_speech)
         self._q = np.power(probas, self.beta).sum(axis=0) + 1.e-8
         return self
 
     def _predict_proba(self, X_context, X_speech):
-        probas = super(PragmaticModel, self)._predict_proba(X_context, X_speech)
-        l_probas = (np.power(probas, self.beta) *
-                    self._p[None, :] / self._q[None, :])
+        probas = super(PragmaticModel, self)._predict_proba(
+            X_context, X_speech)
+        l_probas = (
+            np.power(probas, self.beta) * self._p[None, :] / self._q[None, :])
         return normalize(l_probas, axis=1)
+
+
+class CombinedModel(object):
+    def __init__(self,
+                 vectorizer,
+                 model_generator,
+                 actions,
+                 speech_eps=0.15,
+                 context_eps=0.15):
+
+        self._speech_eps = speech_eps
+        self._context_eps = context_eps
+
+        self.actions = actions
+
+        self.model_generator = model_generator
+        self._vectorizer = vectorizer
+
+        self.context_model = context_model.ContextTreeModel(
+            self.actions, eps=context_eps)
+        self.speech_model = None
```

Comment:
Initialize the speech model here. This would avoid duplication of code in `fit` and `partial_fit`. It will also save you one check in the latter.

Also currently, if the model has already been trained and we call fit again, the context is updated but the speech is only trained on the latest data. We don't want this inconsistance in behavior. If you instead init the speech model at init, you would get an error because the speech model cannot be fit twice, which is what we want (the error).